sqlmesh/core/model/decorator.py

```diff
@@ -82,10 +84,18 @@ def model(
             if statements:
                 common_kwargs[key] = [exp.maybe_parse(s, dialect=dialect) for s in statements]
 
+        if common_kwargs.get("depends_on"):
+            common_kwargs["depends_on"] = {
+                normalize_model_name(name, default_catalog=default_catalog, dialect=dialect)
+                for name in common_kwargs["depends_on"]
+            }
+
         if self.is_sql:
             query = MacroFunc(this=exp.Anonymous(this=entrypoint))
             return create_sql_model(
                 self.name, query, module_path=module_path, dialect=dialect, **common_kwargs
             )
 
-        return create_python_model(self.name, entrypoint, columns=self.columns, **common_kwargs)
+        return create_python_model(
+            self.name, entrypoint, dialect=dialect, columns=self.columns, **common_kwargs
```

Comment:
move dialecgt to common_kwargs so it's overridable

Reply:
Based on your comment it makes me think the intent of this section is to override any keys that are passed into the function with what is defined in the decorator: https://github.com/TobikoData/sqlmesh/blob/52b7048aae19aff72f1bf5441219719e1bda8c11/sqlmesh/core/model/decorator.py#L73-L82

Right now though it doesn't do that and it instead makes sure there are not overlapping keys between what is in the decorator and what is passed into this function. Can you confirm what is the behavior you are expecting for what exists currently?

Reply:
correct, the intent i believe is to have decorator arguments override the default arguments.

Reply:
@tobymao I ended up removing dialect so this feedback is no longer relevant. If we want to change the behavior here then we can do that in a different PR.